sdk/storage/azblob/README.md

```diff
@@ -1,443 +1,276 @@
-handleError(err)
+// authenticate with Azure Active Directory
+cred, err := azidentity.NewDefaultAzureCredential(nil)
+// TODO: handle error
 
-if string(downloadData) != uploadData {
-handleError(errors.New("Uploaded data should be same as downloaded data"))
-}
+// create a client for the specified storage account
+client, err := azblob.NewClient(account, cred, nil)
+// TODO: handle error
 
-if err = reader.Close(); err != nil {
-return
-}
+// blob listings are returned across multiple pages
+pager := client.NewListBlobsFlatPager(containerName, nil)
 
-// ===== 3. List blobs =====
-// List methods returns a pager object which can be used to iterate over the results of a paging operation.
-// To iterate over a page use the NextPage(context.Context) to fetch the next page of results.
-// PageResponse() can be used to iterate over the results of the specific page.
-pager := containerClient.NewListBlobsFlatPager(nil)
+// continue fetching pages until no more remain
 for pager.More() {
-resp, err := pager.NextPage(context.TODO())
-if err != nil {
-handleError(err)
-}
-
-for _, v := range resp.Segment.BlobItems {
-fmt.Println(*v.Name)
-}
+  // advance to the next page
+	page, err := pager.NextPage(context.TODO())
+	// TODO: handle error
+
+	// print the blob names for this page
+	for _, blob := range page.Segment.BlobItems {
+		fmt.Println(*blob.Name)
+	}
 }
-
-// Delete the blob.
-_, err = blockBlobClient.Delete(context.TODO(), nil)
-handleError(err)
-
-// Delete the container.
-_, err = containerClient.Delete(context.TODO(), nil)
-handleError(err)
 ```
 
 ## Troubleshooting
 
-### Error Handling
+All Blob service operations will return an
+[*azcore.ResponseError][azcore_response_error] on failure with a
+populated `ErrorCode` field.  Many of these errors are recoverable.
+The [bloberror][blob_error] package provides the possible Storage error codes
+along with various helper facilities for error handling.
 
-All I/O operations will return an `error` that can be investigated to discover more information about the error. In
-addition, you can investigate the raw response of any response object:
-
-```golang
-import (
-"fmt"
-"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+```go
+const (
+	connectionString = "<connection_string>"
+	containerName    = "sample-container"
 )
 
+// create a client with the provided connection string
+client, err := azblob.NewClientFromConnectionString(connectionString, nil)
+// TODO: handle error
 
-var responseErr *azcore.ResponseError
-errors.As(err, &responseErr)
+// try to delete the container, avoiding any potential race conditions with an in-progress or completed deletion
+_, err = client.DeleteContainer(context.TODO(), containerName, nil)
 
-if responseErr != nil {
-fmt.Println("ErrorCode: " + responseErr.ErrorCode + ", StatusCode: " + strconv.Itoa(responseErr.StatusCode))
-} else {
-fmt.Println(err.Error())
+if respErr := &(azcore.ResponseError{}); errors.As(err, &respErr) {
```

Comment:
Ha, JeffR showed me this trick the other day. It looks odd, but it's nice for scoping.

Reply:
It is nice but I hate it :p.  I want to introduce a helper, `bloberror.Is(err error, codes ...Codes) bool` to simplify this.